index.js

```diff
@@ -33,16 +33,30 @@ export const createHandler = async (client) => {
       const retrievalId = Number(segs[1])
       assert(!Number.isNaN(retrievalId), 400, 'Invalid Retrieval ID')
       const body = await getRawBody(req, { limit: '100kb' })
-      const { success } = JSON.parse(body)
+      const { success, walletAddress } = JSON.parse(body)
+      assert.strictEqual(
+        typeof success,
+        'boolean',
+        400,
+        'boolean .success required'
+      )
+      assert.strictEqual(
+        typeof walletAddress,
+        'string',
+        400,
+        'string .walletAddress required'
+      )
       const { rows } = await client.query(`
         UPDATE retrievals
         SET finished_at = NOW(),
-            success = $1
-        WHERE id = $2
+            success = $2,
+            wallet_address = $3
+        WHERE id = $1
         RETURNING id
       `, [
+        retrievalId,
         success,
-        retrievalId
+        walletAddress
```

Comment:
What is our plan for rejecting duplicate job-result submissions? Again, out of scope of this PR, just want to make sure we don't forget to handle this particular edge case (and an attack vector).

Reply:
Good catch, I forgot that during this PR. Fixed